src/linter_test.cc

```diff
@@ -16,40 +16,80 @@
 
 #include "src/linter.h"
 
-#include "gtest/gtest.h"
+#include <string>
+#include <fstream>
+#include <sstream>
+#include <cstdio>
 
+#include "gtest/gtest.h"
 
-namespace zetasql {
+namespace zetasql::linter {
 
-namespace linter {
+namespace {
 
 TEST(LinterTest, StatementLineLengthCheck) {
-    const absl::string_view sql =
+    absl::string_view sql =
         "SELECT e, sum(f) FROM emp where b = a or c < d group by x";
+    absl::string_view multiline_sql =
+    "SELECT c\n"
+    "some long invalid sql statement that shouldn't stop check\n"
+    "SELECT t from G\n";
 
-    EXPECT_TRUE(checkLineLength(sql).ok());
-    EXPECT_FALSE(checkLineLength(sql, 10).ok());
-    EXPECT_TRUE(checkLineLength(sql, 10, ' ').ok());
-}
+    EXPECT_TRUE(CheckLineLength(sql).ok());
+    EXPECT_FALSE(CheckLineLength(sql, 10).ok());
+    EXPECT_TRUE(CheckLineLength(sql, 10, ' ').ok());
 
+    EXPECT_TRUE(CheckLineLength(multiline_sql).ok());
+    EXPECT_FALSE(CheckLineLength(multiline_sql, 30).ok());
+}
 
 TEST(LinterTest, StatementValidityCheck) {
-    const absl::string_view valid_sql = "SELECT 5+2";
+    EXPECT_TRUE(CheckParserSucceeds("SELECT 5+2").ok());
+    EXPECT_FALSE(CheckParserSucceeds("SELECT 5+2 sss ddd").ok());
 
-    const absl::string_view invalid_sql = "SELECT 5+2 sss ddd";
-
-    EXPECT_TRUE(checkStatement(valid_sql).ok());
-    EXPECT_FALSE(checkStatement(invalid_sql).ok());
-
-    EXPECT_TRUE(checkStatement(
+    EXPECT_TRUE(CheckParserSucceeds(
         "SELECT * FROM emp where b = a or c < d group by x").ok());
 
-    EXPECT_TRUE(checkStatement(
+    EXPECT_TRUE(CheckParserSucceeds(
         "SELECT e, sum(f) FROM emp where b = a or c < d group by x").ok());
```

Comment:
nit: we are not testing ZetaSQL parser, so these two test cases are not really needed. The next one is good thought: it shows that the check works for multiple statements. Let's add another test where the parsing error is in the second statement, but not in the first one. e.g.: `SELECT 1; SELECT 2 3 4;`

Reply:
It currently checks if the whole script is ok or not. So I added test and it failed as expected.